agent/transformers/opentsdb.go

```diff
@@ -0,0 +1,100 @@
+package transformers
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/sensu/sensu-go/types"
+)
+
+// OpenTSDBList contains a list of OpenTSDB metrics
+type OpenTSDBList []OpenTSDB
+
+// OpenTSDB contains values of an OpenTSDB metric
+type OpenTSDB struct {
+	Name      string
+	Value     float64
+	TagSet    []*types.MetricTag
+	Timestamp int64
+}
+
+// Transform transforms metrics in OpenTSDB format to Sensu Metric Format
+func (o OpenTSDBList) Transform() []*types.MetricPoint {
+	var points []*types.MetricPoint
+	for _, metric := range o {
+		mp := &types.MetricPoint{
+			Name:      metric.Name,
+			Value:     metric.Value,
+			Timestamp: metric.Timestamp,
+			Tags:      metric.TagSet,
+		}
+		points = append(points, mp)
+	}
+	return points
+}
+
+// ParseOpenTSDB parses OpenTSDB metrics into a list of OpenTSDB structs
+func ParseOpenTSDB(output string) (OpenTSDBList, error) {
+	openTSDBList := OpenTSDBList{}
+
+	// Split each line of the output into its own metric
+	metrics := strings.Split(output, "\n")
+
+	for _, metric := range metrics {
+		parts := strings.Split(metric, " ")
+
+		// Ensure we have all the required components. A single metric requires a
+		// name, timestamp, value and at least one tag.
+		if len(parts) < 4 {
+			return nil, fmt.Errorf("invalid opentsdb metric: %s", metric)
+		}
+
+		name := parts[0]
+
+		// Convert the timestamp to a unix timestamp with second resolution
+		timestamp, err := strconv.ParseInt(parts[1], 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("invalid opentsdb metric timestamp, must be an integer: %s", parts[1])
+		}
+		if len(parts[1]) == 13 {
```

Comment:
We're diluting the timestamp resolution here right? Can you explain the logic a little? We want to convert the timestamp to a 10 digit timestamp correct?

Reply:
Exactly; opentsdb specs allow both timestamp with second and millisecond precision, so in order to standardize the format, we convert to second resolution if required. That way, handlers know exactly what to expect. Makes sense?